File: network-equipment-pms/pms-frontend/src/api/lowcode-comment.ts
import { del, get, post } from '@/utils/request'

export interface LowCodeComment {
  id?: number
  configType: string
  configId: number
  userId: number
  userName?: string
  content: string
  mentions?: string
  parentId?: number
  createTime?: string
}

/** 评论线程树节点（按 parent_id 构建的多级回复树） */
export interface CommentTreeNode {
  /** 当前评论 */
  comment: LowCodeComment
  /** 子回复列表 */
  replies: CommentTreeNode[]
}

export function getComments(configType: string, configId: number) {
  return get<LowCodeComment[]>('/api/lowcode/comment', { configType, configId })
}

/** 查询线程化评论（按 parent_id 构建树） */
export function getThreadedComments(configType: string, configId: number) {
  return get<CommentTreeNode[]>('/api/lowcode/comment/threaded', { configType, configId })
}

export function addComment(data: LowCodeComment) {
  return post<LowCodeComment>('/api/lowcode/comment', data)
}

export function deleteComment(id: number) {
  return del(`/api/lowcode/comment/${id}`)
}

File: network-equipment-pms/pms-lowcode/src/main/java/com/dp/plat/lowcode/entity/LowCodeComment.java
package com.dp.plat.lowcode.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("pms_lowcode_comment")
public class LowCodeComment {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String configType;
    private Long configId;
    private Long userId;
    private String userName;
    private String content;
    private String mentions; // 逗号分隔的用户 ID
    private Long parentId;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    @TableLogic
    private Integer deleted;
}
